### PR TITLE
An updated dashboard

### PR DESCRIPTION
**Pull request by sschneid**
- unified old `stats-vmpooler-*` scripts
- new graphs
  - weekly clone count by-day
  - weekly clone and boot averages by-day
  - weekly clone count by-platform
- Bootstrap grid layout
  - responsive design optimized for TV / large display
- colors are [hopefully] easier to read
- improved memory footprint
  - removed redundant AJAX calls

Bootstrap, d3, and jQuery all included for speed optimization.

### CODE REVIEW

File: lib/vmpooler/views/dashboard.erb [hunk 1]
```diff
@@ -1,21 +1,63 @@
...
+<div id='header'>
+  <div class='row'>
+    <div id='logo' class='col-md-2'><embed src='/img/logo.gif' style='max-width: 100%; max-height: 100%' /></div>
+    <div id='site_name' class='col-md-10'><%= site_name %></div>
 
-<div id='stats-vmpooler-running' class='module'>
-  <div class='spinner'></div> Loading data...
+    <!-- dashboard-numbers -->
+
+    <div id='dashboard-numbers'></div>
```
Comment (colinPL) [inline lib/vmpooler/views/dashboard.erb]: Above is `site_name` and this is `dashboard-numbers`. Can the use of `_` or `-` be picked and standardized?

Comment (sschneid) [inline lib/vmpooler/views/dashboard.erb]: Good catch... do you have a preference?

File: lib/vmpooler/views/layout.erb [hunk 2]
```diff
@@ -1,35 +1,47 @@
...
 <body>
 
-<div id='wrap'>
-<div id='content'>
-
-<div id='header'>
-  <embed src='/img/logo.gif' width='200px' height='150px' class='logo' />
+  <div id='loading' style='position: absolute; top: 0px; height: 100%; width: 100%; background-color: #fff; z-index: 1;'>
+    <div id='logo' style='position: relative; top: 50%; transform: translateY(-50%);'>
+      <center>
```
Comment (colinPL) [inline lib/vmpooler/views/layout.erb]: My editor says to replace `center` with `<div style="text-align: center;">`. I do not know if its recommendation is valid.

File: lib/vmpooler/views/layout.erb [hunk 3]
```diff
@@ -1,35 +1,47 @@
...
+        <embed src='/img/logo.gif' />
```
Comment (colinPL) [inline lib/vmpooler/views/layout.erb]: Why is this `embed` instead of `img`?

File: lib/vmpooler/views/dashboard.erb (hunk 1 above)
Comment (colinPL) [inline lib/vmpooler/views/dashboard.erb]: I do not. I see plenty of `_` in the committed JavaScript files, so probably that one.

File: lib/vmpooler/views/layout.erb (hunk 2 above)
Comment (rick) [inline lib/vmpooler/views/layout.erb]: If we're moving to `<div>`, it should really move to `<div class="foo">` and have something about `text-align: center` in a CSS style sheet somewhere.  Not sure if we're that inclined to move things, though (i.e., how much of a pain in the butt is that?), right now.

File: lib/vmpooler/public/lib/dashboard.js [hunk 4]
```diff
@@ -0,0 +1,734 @@
...
+    'clone_average': dashboard_data['status']['clone']['duration']['average'] + 's',
+    'capacity': dashboard_data['status']['capacity']['percent'],
+    'total': dashboard_data['status']['queue']['total'],
+    'ready': dashboard_data['status']['queue']['ready'],
+    'cloning': dashboard_data['status']['queue']['cloning'],
+    'booting': dashboard_data['status']['queue']['booting'],
+    'running': dashboard_data['status']['queue']['running'],
+    'completed': dashboard_data['status']['queue']['completed']
+  }
+}
```
Comment (colinPL) [inline lib/vmpooler/public/lib/dashboard.js]: Missing `;`.

File: lib/vmpooler/public/lib/dashboard.js [hunk 5]
```diff
@@ -0,0 +1,734 @@
...
+      }
+
+      running_data['stack'][c][pool_maj] = dashboard_data['running'][pool_maj]['history'][c];
+    }
+  }
+});
+
+if (! running_data['graph']) {
+  running_data['tmp'] = [];
+  for (var metric in running_data['stack']) {
```
Comment (colinPL) [inline lib/vmpooler/public/lib/dashboard.js]: `metric` is the index, not the value. Is this behavior expected?

File: lib/vmpooler/public/lib/dashboard.js [hunk 6]
```diff
@@ -0,0 +1,734 @@
...
+
+var area_boot_avg = d3.svg.area()
+  .interpolate('linear')
+  .x(function(d, i) { return weekly_x(i); })
+  .y0(weekly_height)
+  .y1(function(d) { return weekly_y_boot_avg(d); });
+
+// Create some SVGs
+
+for (graph in graphs = ['clone_count', 'clone_boot_avg']) {
```
Comment (colinPL) [inline lib/vmpooler/public/lib/dashboard.js]:
```
for (graph in graphs = ['clone_count', 'clone_boot_avg']) { console.log(graph); }
> 0
> 1
```

File: lib/vmpooler/public/lib/dashboard.js [hunk 7]
```diff
@@ -0,0 +1,734 @@
...
+    d3.svg.axis()
+      .scale(weekly_x)
+      .ticks(0)
+      .tickSize(-weekly_height)
+      .orient('bottom')
+ );
+
+// Area shapes for clone_count and clone/boot time avgs
+
+var area_clone_count = d3.svg.area()
```
Comment (colinPL) [inline lib/vmpooler/public/lib/dashboard.js]: This name is already defined above.

File: lib/vmpooler/public/lib/dashboard.js [hunk 8]
```diff
@@ -0,0 +1,734 @@
...
+  .interpolate('linear')
+  .x(function(d, i) { return weekly_x(i); })
+  .y0(weekly_height)
+  .y1(function(d) { return weekly_y_clone_count(d); });
+
+var area_boot_avg = d3.svg.area()
```
Comment (colinPL) [inline lib/vmpooler/public/lib/dashboard.js]: This name is already defined above.

**Comment by colinPL** [comment]
I assigned the "needs work" label until the `for (.. in ..)` statements are reviewed and possibly reworked.

File: lib/vmpooler/views/dashboard.erb (hunk 1 above)
Comment (sschneid) [inline lib/vmpooler/views/dashboard.erb]: The `div id`s are all `-`-delimited now; local variables are `_`-delimited.

**Pull request merged**